jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -79,39 +79,47 @@
 	@Deprecated // this should be deleted in 7.0.x
 	private boolean deprecatedIdStrategy = Boolean.getBoolean("jbpm.v5.id.strategy");
 	private AtomicLong singleNodeInstanceCounter = new AtomicLong(0);
-	
+
 	private Map<String, List<EventListener>> eventListeners = new HashMap<String, List<EventListener>>();
 	private Map<String, List<EventListener>> externalEventListeners = new HashMap<String, List<EventListener>>();
+
 	private List<String> completedNodeIds = new ArrayList<String>();
 	private List<String> activatingNodeIds;
 	private Map<String, Integer> iterationLevels = new HashMap<String, Integer>();
 	private int currentLevel;
 	private boolean persisted = false;
 	private Object faultData;
-	
+
 	private boolean signalCompletion = true;
 
+	public static final EventListener EMPTY_EVENT_LISTENER = new EventListener() {
```

Comment:
@mrietveld what I mean is to move this EventListener to a public class (separate file) that will have only single instance and shall be accessed via static method only (so private constructor only) and then referenced from both WorkflowProcessInstanceImpl and EventNode which will give use no dependency of CompositeNodeInstance and EventNodeInstance to WorkflowProcessInstanceImpl

though this is just a minor thing